### PR TITLE
eliminate PhiRot_Type from StateData

### PR DESCRIPTION
**Pull request by zingale**
## PR checklist

- [ ] test suite needs to be run on this PR
- [ ] this PR will change answers in the test suite to more than roundoff level
- [ ] all newly-added functions have docstrings as per the coding conventions
- [ ] the `CHANGES` file has been updated, if appropriate
- [ ] if appropriate, this change is described in the docs

### CODE REVIEW

**Comment by zingale** [comment]
still need to do wdmerger

**Comment by zingale** [comment]
tests: http://groot.astro.sunysb.edu/Castro/test-suite/gfortran/2021-09-09-002/index.html

File: Source/scf/scf_relax.cpp [hunk 1]
```diff
@@ -620,7 +610,8 @@ Castro::do_hscf_solve()
 
         for (int lev = 0; lev <= finest_level; ++lev) {
 
-            const Real* dx = parent->Geom(lev).CellSize();
+            auto geomdata = parent->Geom(lev).data();
+            const Real* dx = geomdata.CellSize();
```
Comment (maxpkatz) [inline Source/scf/scf_relax.cpp:614]: This should probably be CellSizeArray for safety.

**Comment by maxpkatz** [comment]
It would be consistent with past behavior if this increased the checkpoint version number, although I am not sure if it's necessary.

**Comment by zingale** [comment]
I'll increment the checkpoint number just to be consistent

File: Source/scf/scf_relax.cpp (hunk 1 above)
Comment (zingale) [reply Source/scf/scf_relax.cpp]: done

**Comment by zingale** [comment]
updated with checkpoint version increment

**Review by maxpkatz — APPROVED** [review]

**Pull request merged**